internal/testtypes/builders.go

```diff
@@ -0,0 +1,530 @@
+	c.infrastructureClusterTemplate = t
+	return c
+}
+
+// WithControlPlaneTemplate adds the passed ControlPlaneTemplate to the ClusterClassBuilder.
+func (c *ClusterClassBuilder) WithControlPlaneTemplate(t *unstructured.Unstructured) *ClusterClassBuilder {
+	c.controlPlaneTemplate = t
+	return c
+}
+
+// WithControlPlaneMetadata adds the given labels and annotations for use with the ControlPlane to the ClusterClassBuilder.
+func (c *ClusterClassBuilder) WithControlPlaneMetadata(labels, annotations map[string]string) *ClusterClassBuilder {
+	c.controlPlaneMetadata = &clusterv1.ObjectMeta{
+		Labels:      labels,
+		Annotations: annotations,
+	}
+	return c
+}
+
+// WithControlPlaneInfrastructureMachineTemplate adds the ControlPlane's InfrastructureMachineTemplate to the ClusterClassBuilder.
+func (c *ClusterClassBuilder) WithControlPlaneInfrastructureMachineTemplate(t *unstructured.Unstructured) *ClusterClassBuilder {
+	c.controlPlaneInfrastructureMachineTemplate = t
+	return c
+}
+
+// WithWorkerMachineDeploymentClass adds the variables and objects needed to create MachineDeploymentTemplates for a ClusterClassBuilder.
+func (c *ClusterClassBuilder) WithWorkerMachineDeploymentClass(mdcs []clusterv1.MachineDeploymentClass) *ClusterClassBuilder {
+	if c.machineDeploymentClasses == nil {
+		c.machineDeploymentClasses = make([]clusterv1.MachineDeploymentClass, 0)
+	}
+	c.machineDeploymentClasses = append(c.machineDeploymentClasses, mdcs...)
+	return c
+}
+
+// Build takes the objects and variables in the ClusterClass builder and uses them to create a ClusterClass object.
+func (c *ClusterClassBuilder) Build() *clusterv1.ClusterClass {
+	obj := &clusterv1.ClusterClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterClass",
+			APIVersion: clusterv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.name,
+			Namespace: c.namespace,
+		},
+		Spec: clusterv1.ClusterClassSpec{},
+	}
+	if c.infrastructureClusterTemplate != nil {
+		obj.Spec.Infrastructure = clusterv1.LocalObjectTemplate{
+			Ref: objToRef(c.infrastructureClusterTemplate),
+		}
+	}
+	if c.controlPlaneMetadata != nil {
+		obj.Spec.ControlPlane.Metadata = *c.controlPlaneMetadata
+	}
+	if c.controlPlaneTemplate != nil {
+		obj.Spec.ControlPlane.LocalObjectTemplate = clusterv1.LocalObjectTemplate{
+			Ref: objToRef(c.controlPlaneTemplate),
+		}
+	}
+	if c.controlPlaneInfrastructureMachineTemplate != nil {
+		obj.Spec.ControlPlane.MachineInfrastructure = &clusterv1.LocalObjectTemplate{
+			Ref: objToRef(c.controlPlaneInfrastructureMachineTemplate),
+		}
+	}
+	if len(c.machineDeploymentClasses) > 0 {
+		obj.Spec.Workers.MachineDeployments = append(obj.Spec.Workers.MachineDeployments, c.machineDeploymentClasses...)
+	}
+	// TODO: This resource version exists because of the way the fakeclient handlies this internally. Need custom matcher to remove this.
+	obj.ResourceVersion = "999"
```

Comment:
It is required to remove this, otherwise it wont't be possible to use this builders with testenv
(same for other builders)